src/NuGet.Indexing/NuGetSearcherManager.cs

```diff
@@ -122,8 +122,12 @@ public class NuGetSearcherManager : SearcherManager<NuGetIndexSearcher>
             var searcherManager = new NuGetSearcherManager(logger, indexProvider, loader, config.AuxiliaryDataRefreshRateSec, config.IndexReloadRateSec);
 
             var registrationBaseAddress = config.RegistrationBaseAddress;
-            searcherManager.RegistrationBaseAddress["http"] = MakeRegistrationBaseAddress("http", registrationBaseAddress);
-            searcherManager.RegistrationBaseAddress["https"] = MakeRegistrationBaseAddress("https", registrationBaseAddress);
+            var semVer2RegistrationBaseAddress = config.SemVer2RegistrationBaseAddress;
+
+            searcherManager.RegistrationBaseAddresses.HttpRegistrationAddress = MakeRegistrationBaseAddress("http", registrationBaseAddress);
```

Comment:
nit: I think this logic would make more sense inside the constructor for `RegistrationAddresses`.